### ISSUE TITLE
Ścieżka specjalizacyjna w programowaniu: C++, C#, Java

### ISSUE THREAD

**Issue by SMeszynski** [IS-UMK/sylabusy#4]
# Ścieżka specjalizacyjna w programowaniu: C++, C#, Java

Celem jest opracowanie ścieżki specjalizacyjnej, umożliwiającej studentom rozwój w wybranym języku programowania: C++, C#, lub Java. Każda ścieżka obejmuje trzy semestry nauki, w których stopniowo realizowane są coraz bardziej zaawansowane zagadnienia. Struktura zajęć jest uniwersalna dla wszystkich trzech języków, z uwzględnieniem ich specyficznych różnic i możliwości. 

## Semestr 1: Podstawy programowania
**Cel:** Opanowanie podstaw wybranego języka programowania, w tym składni, struktur kontrolnych, funkcji oraz wskaźników (w C++). 

1. Wprowadzenie do wybranego języka
2. Podstawy składni i typów danych
3. Kontrola przepływu programu
4. Funkcje
5. Podstawy struktur danych
6. Obsługa wyjątków
7. Testy jednostkowe

## Semestr 2: Programowanie obiektowe
**Cel:** Wprowadzenie do paradygmatu obiektowego z uwzględnieniem różnic między językami, takich jak użycie klas, dziedziczenie, polimorfizm i wzorce projektowe.

1. Podstawy obiektowości
2. Dziedziczenie i polimorfizm
3. Interfejsy i abstrakcja
4. Przeciążanie funkcji i operatorów
5. Obsługa plików
6. Podstawowe wzorce projektowe

## Semestr 3: Zaawansowane techniki i projekty
**Cel:** Nauka zaawansowanych technik programowania w wybranym języku.

1. Zaawansowane struktury danych
2. Programowanie wielowątkowe
3. Szablony/generics
4. Biblioteki i frameworki
5. Wprowadzenie do nowoczesnych technik

**Comment by lemkis** [comment]
Czyli dobrze rozumiem że to 9 przedmiotów? 3 stopnie zaawansowania dla każdego z języków c++ c# Java?

**Comment by lemkis** [comment]
Czy podejmujesz się prowadzenia tych wszystkich przedmiotów? Trochę dużo jak na 1 osobę

**Comment by SMeszynski** [comment]
Oczywiście, jak już wielokrotnie wspominałem, jest to jedynie propozycja dotycząca struktury nauczania w programowaniu obiektowym, zakładająca trzy ścieżki w zależności od preferencji studentów. Zdaję sobie sprawę, że dla jednej osoby to zbyt wiele do udźwignięcia. Być może warto rozważyć połączenie sił między automatykami i informatykami w zakresie nauczania C++? To mogłoby odciążyć prowadzących i zapewnić spójniejsze podejście do tematu.
